### DISCUSSION TITLE
Datetime scraping in XElement

### DISCUSSION THREAD

**Discussion by Mlgprokn** [VerifyTests/Verify#771] [Q&A] [answered]
Hello,

I am working with XElements and one of the Nodes has an Attribute for date and time, which is always set to DateTime.Now(). I saw in the documentation that I need to use "DateTime_1" as the value in the verified file. However, this does not seem to work. Can someone help me with this, I am attaching a screenshot of the failed test.

![image](https://user-images.githubusercontent.com/46377468/214534968-1611e4c9-4c84-4df5-b344-6bc659e7769b.png)

**Comment by SimonCropp** [comment, marked as answer]
thats a bug. can u try v 19.8.2

**Reply by Mlgprokn** [reply]
Thank you!

**Reply by uuuus** [reply]
In 19.8.2 I have the same issue.

**Reply by SimonCropp** [reply]
@uuuus i will need a repro

**Comment by uuuus** [comment]
I get file:
{
  BirthDate: DateTime_1,
  Name: Test
}
after running test:
[UsesVerify]
    public class DateTimeTest
    {
        class Person
        {
            public DateTime BirthDate { get; set; }
            public string Name { get; set; }
        }

        [Fact]
        public async Task BirthDateShouldSerializeAsDateTime()
        {
            var p = new Person { Name = "Test", BirthDate = new
DateTime(1980, 1,  1) };
            await Verifier.Verify(p);
        }
    }

**Reply by SimonCropp** [reply]
@uuuus thats how it should be working. if u want to disable date scrubbing see https://github.com/VerifyTests/Verify/blob/main/docs/serializer-settings.md#snippet-dontscrubdatetimesfluent

**Comment by uuuus** [comment]
I see.
Is there any single setting to be sure that complete object is serialized
with all original values?

I have class with property:
public int Value { get; set; }

When Value = 0 property is not serialized.

**Reply by SimonCropp** [reply]
```
    [ModuleInitializer]
    public static void Init()
    {
        VerifierSettings.DontScrubGuids();
        VerifierSettings.DontSortDictionaries();
        VerifierSettings.DontIgnoreEmptyCollections();
        VerifierSettings.DontScrubDateTimes();
        VerifierSettings.DontScrubProjectDirectory();
        VerifierSettings.DontScrubSolutionDirectory();
        VerifierSettings.AddExtraSettings(
            _ =>
            {
                _.DefaultValueHandling = DefaultValueHandling.Include;
                _.NullValueHandling = NullValueHandling.Include;
            });
    }
```
